iroha/src/wsv.rs

```diff
@@ -26,43 +24,103 @@ impl WorldStateView {
     pub fn new(world: World) -> Self {
         WorldStateView {
             world,
-            transactions: HashMap::new(),
             config: Configuration::default(),
+            blocks: Vec::new(),
         }
     }
 
     /// [`WorldStateView`] constructor with configuration.
-    pub fn from_config(config: Configuration, world: World) -> Self {
+    pub const fn from_config(config: Configuration, world: World) -> Self {
         WorldStateView {
             world,
-            transactions: HashMap::new(),
+            blocks: Vec::new(),
             config,
         }
     }
 
+    fn iter_block_tx_hash(block: &'_ VersionedCommittedBlock) -> impl Iterator<Item = Hash> + '_ {
+        let v1_block = block.as_inner_v1();
+        v1_block
+            .rejected_transactions
+            .iter()
+            .map(VersionedRejectedTransaction::hash)
+            .chain(
+                v1_block
+                    .transactions
+                    .iter()
+                    .map(VersionedValidTransaction::hash),
+            )
+    }
+
+    fn iter_block_tx(
+        block: &'_ VersionedCommittedBlock,
+    ) -> impl Iterator<Item = TransactionValue> + '_ {
+        let v1_block = block.as_inner_v1();
+        v1_block
+            .rejected_transactions
+            .iter()
+            .cloned()
+            .map(TransactionValue::RejectedTransaction)
+            .chain(
+                v1_block
+                    .transactions
+                    .iter()
+                    .cloned()
+                    .map(VersionedAcceptedTransaction::from)
+                    .map(Into::into)
+                    .map(TransactionValue::Transaction),
+            )
+    }
```

Comment:
No need to move it out into a separate function it is only needed in one place. Let's move it there.

Reply:
It is not about repetition, but about modularity. Seems that it can be used later on. May be we should even move to `VersionedCommittedBlock` itself.

Reply:
Yes, I think you can move it into `VersionCommittedBlock::iter_transaction_values`